src/NServiceBus.Transport.Msmq/MessagePump.cs

```diff
@@ -175,12 +197,14 @@ async Task PumpMessages(CancellationToken messagePumpCancellationToken)
 
                     messagePumpCancellationToken.ThrowIfCancellationRequested();
 
-                    await concurrencyLimiter.WaitAsync(messagePumpCancellationToken).ConfigureAwait(false);
+                    var localLimiter = concurrencyLimiter;
+
+                    await localLimiter.WaitAsync(messagePumpCancellationToken).ConfigureAwait(false);
 
                     _ = Task.Factory.StartNew(state =>
                         {
                             var (messagePump, cancellationToken) = ((MessagePump, CancellationToken))state;
-                            return ReceiveMessagesSwallowExceptionsAndReleaseConcurrencyLimiter(messagePump, cancellationToken);
+                            return ReceiveMessagesSwallowExceptionsAndReleaseConcurrencyLimiter(messagePump, localLimiter, cancellationToken);
```

Comment:
This is now capturing a closure which was previously avoided. To avoid the closure, you need to pass in the limiter via the value tuple and extract it inside the `StartNew`

But see below comment. Maybe this is not needed

Reply:
Thanks, I'll fix it.

Reply:
👍